core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1596,7 +1596,33 @@ protected Scope visitQuery(Query node, Optional<Scope> scope)
                     .withRelationType(RelationId.of(node), queryBodyScope.getRelationType())
                     .build();
 
+            // collect output columns info
+            ImmutableList.Builder<OutputColumn> outputColumnsBuilder = ImmutableList.builder();
+            for (Field field : queryScope.getRelationType().getVisibleFields()) {
+                OutputColumn outputColumn = new OutputColumn(new Column(field.getName().orElse(""), field.getType().toString()), analysis.getSourceColumns(field));
+                outputColumnsBuilder.add(outputColumn);
+            }
+
             analysis.setScope(node, queryScope);
+            ImmutableList<OutputColumn> outputColumns = outputColumnsBuilder.build();
+            if (!outputColumns.isEmpty()) {
+                QualifiedObjectName qualifiedName = new QualifiedObjectName("", "", "");
+                CatalogHandle.CatalogVersion version = new CatalogHandle.CatalogVersion("1");
+                if (node.getQueryBody() instanceof QuerySpecification querySpecification) {
```

Comment:
This check is a bit brittle as we capture them only for simple queries and not for queries which has `WITH` clause in the begining.